pkg/dockerregistry/server/manifestservice.go

```diff
@@ -47,38 +40,29 @@ func (m *manifestService) Exists(ctx context.Context, dgst digest.Digest) (bool,
 func (m *manifestService) Get(ctx context.Context, dgst digest.Digest, options ...distribution.ManifestServiceOption) (distribution.Manifest, error) {
 	context.GetLogger(ctx).Debugf("(*manifestService).Get")
 
-	image, _, _, err := m.repo.getStoredImageOfImageStream(dgst)
```

Comment:
This moves the status of "source of truth" from etcd to registry's storage. Is that intentional?

Reply:
Yes, it fixes #9550.

Reply:
IMHO etcd should stay as a source of truth. Deleting an image using `oc delete` should prevent registry from serving it, which applies to deleting imagestreamtag or whole imagestream. Tagging the image into new image stream using `oc tag` should allow registry to serve it from the corresponding repository.

Moving the source of truth from etcd to storage implies a need for one-way synchronization of storage state to etcd. We don't have that in place yet.

Reply:
OK, you are right, this code has a bug. Image streams should be the source of truth for tagged manifests. Though, image streams don't contain information about untagged manifests and can't be used as a source of truth for them.

Reply:
> Though, image streams don't contain information about untagged manifests and can't be used as a source of truth for them.

Well, we don't keep the history of image presence in image streams (something like untagged images). And I don't think it's desirable since image stream is already too fat.

The rejection to serve a manifest can be determined from the absence of image reference in the image stream.

Reply:
I don't want to get a rejection, I want to pull it back even if I don't tag it.